### PR TITLE
Remove Unneeded ATDM Configurations

### PR DESCRIPTION
**Pull request by ZUUL42**
@trilinos/framework 

## 

This will remove the configurations identified as no longer needed in preparation for the ATDM Migration.

These are the configurations marked as "Not Needed":
System | Configuration | --
-- | -- | --
ats1 (mutrino) | ats1-knl_intel-18.0.5_mpich-7.7.6_openmp_static_opt
-- | ats1-knl_intel-18.0.5_mpich-7.7.6_openmp_static_dbg
-- | cts1 (serrano, eclipse, ghost)
-- | cts1-intel-18.0.2_openmpi-4.0.3_openmp_static_dbg
-- | cts1-intel-18.0.2_openmpi-4.0.3_openmp_static_opt
mutrino | intel-debug-openmp-HSW | system removed completely
-- | intel-debug-openmp-KNL
-- | intel-opt-openmp-HSW
-- | intel-opt-openmp-KNL
-- | intel-complex-openmp-opt-KNL
ride (or white) | cuda-9.2-gnu-7.2.0-debug | system removed completely
-- | cuda-9.2-gnu-7.2.0-release
-- | cuda-9.2-gnu-7.2.0-release-debug
-- | gnu-7.2.0-openmp-debug
-- | gnu-7.2.0-openmp-release
-- | gnu-7.2.0-openmp-release-debug
-- | cuda-10.1-gnu-7.2.0-debug
-- | cuda-10.1-gnu-7.2.0-release
-- | cuda-10.1-gnu-7.2.0-release-debug
sems-rhel6 | sems-rhel6-gnu-7.2.0-openmp-debug | system removed completely
-- | sems-rhel6-gnu-7.2.0-openmp-release
-- | sems-rhel6-gnu-7.2.0-openmp-release-debug
-- | sems-rhel6-gnu-7.2.0-openmp-complex-shared-release-debug
-- | sems-rhel6-intel-17.0.1-openmp-release
sems-rhel7 | sems-rhel7-intel-18.0.5-openmp-complex-shared-release-debug
shiller (or hansen) | cuda-8.0-debug | system removed completely
-- | cuda-8.0-opt
-- | cuda-9.0-debug
-- | cuda-9.0-opt
-- | gnu-debug-openmp
-- | gnu-debug-serial
-- | gnu-opt-openmp
-- | gnu-opt-serial
-- | intel-debug-openmp
-- | intel-debug-serial
-- | intel-opt-openmp
-- | intel-opt-serial
spack-rhel | spack-rhel-clang-5.0.1-openmpi-1.10.2-openmp-shared-release-debug | system removed completely
-- | spack-rhel-clang-5.0.1-openmpi-1.10.2-openmp-static-release-debug
-- | spack-rhel-clang-5.0.1-openmpi-1.10.2-serial-shared-release-debug
-- | spack-rhel-clang-5.0.1-openmpi-1.10.2-serial-static-release-debug
-- | spack-rhel-gnu-7.2.0-opemmpi-1.10.1-openmp-shared-release-debug
-- | spack-rhel-gnu-7.2.0-opemmpi-1.10.1-openmp-static-release-debug
-- | spack-rhel-gnu-7.2.0-opemmpi-1.10.1-serial-shared-release-debug
-- | spack-rhel-gnu-7.2.0-opemmpi-1.10.1-serial-static-release-debug
-- | spack-rhel-gnu-7.2.0-opemmpi-2.1.2-openmp-shared-release-debug
-- | spack-rhel-gnu-7.2.0-opemmpi-2.1.2-serial-shared-release-debug

Complete list, needed and not, on Confluence.

## Stakeholder Feedback

Ensure no ATDM application customers need the environment:

- EMPIRE: Confirmed this list is OK for cleanup
- SPARC: "Can you clarify the ones in the mutrino group? Which Intel, MPI, etc.?"
- GEMMA: Haven't heard from.

### CODE REVIEW

**Review by bartlettroscoe — COMMENTED** [review]
I would like to hold off on having the Trilinos Framework team on making any changes to the ATDM Trilinos configuration until they have taken over triaging the ATDM Trilinos builds and interfacing with the customers.

Also, note that all of the ATS-1 builds are currently broken as described in https://sems-atlassian-srn.sandia.gov/browse/ATDV-409.

I would like to move the 'spack-rhel' configuration to the 'contributed' folder instead of deleting it altogether.

**Comment by jmgate** [comment]
@ZUUL42, can you revert 87cddadb424503d328a2473f2a503108b0a01195.  We're just wanting to remove the systems that are no longer supported, not any additional configurations that apps indicated they didn't care about.  That can come later.

**Review by bartlettroscoe — CHANGES_REQUESTED** [review]
Note that if one removes these systems, one also has to remove them from the README.md files:

* cmake/std/atdm/README.md
* cmake/ctest/drivers/atdm/README.md

**Comment by jmgate** [comment]
@ZUUL42, there are also corresponding directories to remove in `cmake/ctest/drivers/atdm`.

**Comment by jmgate** [comment]
@ZUUL42, where do things stand here?

**Comment by ZUUL42** [comment]
@jmgate @bartlettroscoe
I reverted the individual removals and made the associated ctest system removals as well as the references in the README.md files. I did leave a few references in the READMEs as they appear to be used as examples names that don't necessarily need to be pulled.

Let me know if that seems about where we want to be.

**Review by jmgate — CHANGES_REQUESTED** [review]
Remove the changes to the "CEE RHEL6 and RHEL7 Environment" section of the README.md, and then this looks fine to me.  The environment name is still `cee-rhel6`, even if RHEL6 is no longer supported.  Perhaps it should be updated, but that's a separate PR, and we probably don't want to bother.

'Twould be good to get @bartlettroscoe's eyes on this again before merging.

File: cmake/ctest/drivers/atdm/README.md
Comment (bartlettroscoe) [inline cmake/ctest/drivers/atdm/README.md:473]: This should really be removed too.  This has not been used in a long time.

File: cmake/std/atdm/README.md
Comment (bartlettroscoe) [inline cmake/std/atdm/README.md:987]: Yes, you have to put back this section "CEE RHEL6 and RHEL7 Environment".  Note the need to change the name from 'cee-rhel6' to 'cee-rhel7':

* https://sems-atlassian-srn.sandia.gov/browse/ATDV-391

Can someone in the @trilinos/framework team work that?  Note this will also require coordination with the SPARC team as well.

File: cmake/std/atdm/README.md
Comment (bartlettroscoe) [inline cmake/std/atdm/README.md:990]: No, the name 'cee-rhel6' has not been changed to 'cee-rhel7' yet.  That is the work of:

* https://sems-atlassian-srn.sandia.gov/browse/ATDV-391

File: cmake/std/atdm/README.md
Comment (bartlettroscoe) [inline cmake/std/atdm/README.md:651]: Why is this being removed?  As long as the Trilinos PR tester is running on 'ride', this configuration should stay as it is currently running and posting to CDash.

Please don't touch anything that impacts the ATDM Trilinos builds posting to CDash until the Framework team has taken over full responsibility for these builds.

**Review by bartlettroscoe — CHANGES_REQUESTED** [review]
Needs a few things:

1. Please put back the section '### CEE RHEL6 and RHEL7 Environment' exactly the way it was.  The name changes from 'cee-rhel6' to 'cee-rhel7' will need to be done in a different story and must be coordinated with the SPARC developers.   See https://sems-atlassian-srn.sandia.gov/browse/ATDV-391

2. Please undelete the 'ride/white' env.  There are still ATDM Trilinos builds for this posting to CDash and they include the only CUDA RDC builds currently being run.   (If you want to delete that env, someone will need to duplicate the builds on the 'ats2' machine 'vortex' first.)

In this PR, please don't change anything that will impact any of the current ATDM Trilinos builds currently being posted to CDash as shown [here](https://testing.sandia.gov/cdash/index.php?project=Trilinos&filtercount=1&showfilters=1&field1=buildname&compare1=65&value1=Trilinos-atdm-).

Until the Framework team takes over the maintenance of these builds, please don't make any changes that could break them or break them for customers.

File: cmake/ctest/drivers/atdm/README.md
Comment (jmgate) [reply cmake/ctest/drivers/atdm/README.md:473]: I'm not seeing a `sems_gcc-7.2.0` directory here.  Was it perhaps removed long ago, and we just need to remove the corresponding lines in the `README.md`?

File: cmake/std/atdm/README.md
Comment (jmgate) [reply cmake/std/atdm/README.md:987]: > Can someone in the @trilinos/framework team work that?

No, we'll likely just leave it as-is.

File: cmake/std/atdm/README.md
Comment (jmgate) [reply cmake/std/atdm/README.md:651]: When speaking with EMPIRE, GEMMA, and SPARC, they all indicated they have no need of the ride environments/configurations.  If you prefer, though, we can add the ride directories back for the time being.

**Comment by jmgate** [comment]
@trilinos/framework, just curious, but do we have an ETA on the move to vortex?

**Review by bartlettroscoe — CHANGES_REQUESTED** [review]
@ZUUL42,

Instead of deleting it, can you please move the `cmake/std/atdm/spack-rhel/` directory to `cmake/std/adm/contributed/sparck-rhel/` and move the text for it in `cmake/std/atdm/README.md` into a new file `cmake/std/adm/contributed/sparck-rhel/README.md`?  That way, I can keep history for these files and it gets out of the way of the officially supported systems.

Thanks

**Comment by ZUUL42** [comment]
@jmgate @bartlettroscoe
Let me know if the last set of changes get things into an acceptable state.
I brought back `ride/white` and `spack-rhel`, moved `spack-rhel` to `contribute`, and reversed the cee-rhel6 changes.

File: cmake/ctest/drivers/atdm/README.md
Comment (bartlettroscoe) [inline cmake/ctest/drivers/atdm/README.md:473]: We can get rid of 'serrano' too.  That is not a directory here anymore.

File: cmake/std/atdm/ride/all_supported_builds.sh
Comment (bartlettroscoe) [inline cmake/std/atdm/ride/all_supported_builds.sh:12]: No, you have to put these back while we are still running builds on 'ride'.  As long as 'ride' builds are posting to CDash, those builds have to be listed here.

**Review by bartlettroscoe — CHANGES_REQUESTED** [review]
Need to put back the supported 'ride' builds for now.  (Until we can duplicate the 'RDC' builds on other platforms, for example.  And there is also the issue of matching up the Trilinos PR CUDA builds and the ATDM Trilinos CUDA builds.  We need the 'ride' builds to do that currently to compare apples to apples.)

We can also remove documentatation for the 'seranno' directory as well (see note below).

**Review by bartlettroscoe — APPROVED** [review]
@ZUUL42, with my latest commits https://github.com/trilinos/Trilinos/pull/8362/commits/0a8ce54871e2e944856116219a6da787799aa47f (resolved trivial merge conflict) and https://github.com/trilinos/Trilinos/pull/8362/commits/075247c5493b2d5745332eb2041e6e89dd85dd63 (removed some more unneeded files), I think this is ready to merge.

**Comment by jmgate** [comment]
Interesting.  Apparently GitHub's implementation of draft vs ready for review isn't controlled by the presence of "DRAFT:" in the PR title.  Let's see how the autotester does, and then we should be good to merge.  Thanks for all the help, @bartlettroscoe, and thanks, @ZUUL42, for taking care of this.

**Comment by bartlettroscoe** [comment]
@jmgate, I requested a new review.  (Your last review was requesting changes.)

**Comment by jmgate** [comment]
Can't seems to approve&mdash;GitHub glitching out on me.  As far as I'm concerned, feel free to merge.

**Comment by bartlettroscoe** [comment]
> Can't seems to approve—GitHub glitching out on me. As far as I'm concerned, feel free to merge.

@jmgate, @ZUUL42, I think someone just needs to put 'AT: AUTOMERGE' on this PR then.  I guess I will go ahead and do that.

**Comment by bartlettroscoe** [comment]
I added the 'AT: AUTOMERGE' label so hopefully the autotester will do the merge.

I am wondering if the autotester will allow the merge given that I re-requested @jmgate's review and now it shows the only active review being mine with an 'approved'.

**Comment by allevin** [comment]
> I added the 'AT: AUTOMERGE' label so hopefully the autotester will do the merge.
> 
> I am wondering if the autotester will allow the merge given that I re-requested @jmgate's review and now it shows the only active review being mine with an 'approved'.

@jmgate @jwillenbring @bartlettroscoe The autotester is trying (and failing) to merge...  It is receiving errors from Github saying it cannot merge because "1 review requesting changes and 1 approving review by reviewers with write access.'

According to the info below, the "Merge pull request" button is disabled because there is 1 change requested by @jmgate.  I believe that @jmgate 's review must be addressed before this PR will go forward.

**Comment by jmgate** [comment]
Cool.  I was curious if the autotester would catch that.  Glad to hear that it did.  I’ll see if I can approve on my iPad.  No idea why my laptop is having issues with GitHub these days.

**Review by jmgate — APPROVED** [review]

**Comment by bartlettroscoe** [comment]
Related to:
* SEPW-215

**Pull request merged**
